src/driver.rs

```diff
@@ -0,0 +1,190 @@
+// error-pattern:yummy
+#![feature(box_syntax)]
+#![feature(rustc_private)]
+#![allow(unknown_lints, missing_docs_in_private_items)]
+
+extern crate clippy_lints;
+extern crate getopts;
+extern crate rustc;
+extern crate rustc_driver;
+extern crate rustc_errors;
+extern crate rustc_plugin;
+extern crate syntax;
+
+use rustc_driver::{driver, Compilation, CompilerCalls, RustcDefaultCalls};
+use rustc::session::{config, CompileIncomplete, Session};
+use rustc::session::config::{ErrorOutputType, Input};
+use std::path::PathBuf;
+use std::process::Command;
+use syntax::ast;
+
+struct ClippyCompilerCalls {
+    default: RustcDefaultCalls,
+    run_lints: bool,
+}
+
+impl ClippyCompilerCalls {
+    fn new(run_lints: bool) -> Self {
+        Self {
+            default: RustcDefaultCalls,
+            run_lints: run_lints,
+        }
+    }
+}
+
+impl<'a> CompilerCalls<'a> for ClippyCompilerCalls {
```

Comment:
Was any code changed here? It looks ok to me, but it's unclear what was added/removed.

Reply:
No. I should have probably duplicated the file in a separate commit before removing things...